### ISSUE TITLE
Override a theme's sass

### ISSUE THREAD

**Issue by lightclient** [getzola/zola#837]
As far as I can tell, only files in the `template` & `static` folders can be overridden. It would be great if we could also override a theme's stylesheet though. I came across [this](https://zola.discourse.group/t/overriding-sass-in-tempate/175) thread where @Keats mentions how this might be resolved:

> Sass loading would be very complex to do, essentially rewriting imports in .scss files manually. I believe if sass-rs was supporting sass importers it would be trivial to do but it isn’t implemented.

It seems like this is something that would pretty valuable to `zola` users!

**Comment by Keats** [comment]
That would be an issue for [sass-rs](https://github.com/compass-rs/sass-rs) though, not Zola. Since I'm also the maintainer of sass-rs, I know I won't work on this feature myself so it would have to come from someone else.

**Comment by lightclient** [comment]
Good to know! I may be interested in tackling this. Would you be able to give me a ~paragraph on how you might go about implementing this feature?

**Comment by Keats** [comment]
It's mostly figuring out what FFI to use in https://github.com/compass-rs/sass-rs/blob/master/sass-sys/src/bindings.rs and them in https://github.com/compass-rs/sass-rs/tree/master/src/bindings

I am guessing the easiest solution is to look at what [node-sass](https://github.com/sass/node-sass) is doing for that.
Something important to remember is that you would still need to copy/paste the whole Sass file from the theme in your own sass folder, it isn't possible to just change some rules (I think; I never actually used Sass importers in my life). I think the actual first step should be to try with `node-sass` if the overriding works as expected before doing anything in sass-rs.

**Comment by lightclient** [comment]
Thanks @Keats! I'll look into it.

**Comment by Folyd** [comment]
Hi @lightclient. I also need to override the sass file of a theme.  It's possible to do this today?

**Comment by lightclient** [comment]
@Folyd I'm not certain, I ended up building a static site from scratch.

**Comment by Keats** [comment]
It's not possible right now.

**Comment by Wabuo** [comment]
What's the current stand of affairs, can we by now somehow override a theme's sass?

I would love to change the -link-hover-color without hard copying all the sass files.

What's the best practise for a case like that?

**Comment by Keats** [comment]
No changes right now. The issue is how it would look like in practice.

**Comment by jhpratt** [comment]
CSS cascade layers is probably the simplest solution here. It's been available on all major browsers for just about two years now, so relying on it seems reasonable.

**Comment by Keats** [comment]
I think themes using CSS variables would already be a huge improvement

**Comment by jhpratt** [comment]
Of course, but that's an uphill battle. Cascade layers can be handled without any changes to the themes whatsoever, given that they can be imported in a specified layer.
